### PR TITLE
Implement model visibility toggling

### PR DESCRIPTION
**Pull request by Kevinskwk**
This PR adds the feature of hiding & showing the static models on each level in the toggle_floors plugin. Static model names are recorded in the plugin sdf specification and hide/show messages are published on click of the toggle floor button.
Screenshot from 2020-08-31 17-56-53
Screenshot from 2020-08-31 17-57-01

### CODE REVIEW

File: building_sim_plugins/building_gazebo_plugins/src/toggle_floors.cpp [hunk 1]
```diff
@@ -78,6 +90,11 @@ class ToggleFloors : public gazebo::GUIPlugin
     visual_msg.set_name(model_name);
     visual_msg.set_visible(visible);
     visual_pub->Publish(visual_msg);
+    for (string model : models)
```
Comment (Yadunund) [inline building_sim_plugins/building_gazebo_plugins/src/toggle_floors.cpp:93]:
```suggestion
    for (const string& model : models)
```

File: building_sim_plugins/building_gazebo_plugins/src/toggle_floors.cpp [hunk 2]
```diff
@@ -45,6 +45,17 @@ class ToggleFloors : public gazebo::GUIPlugin
       string floor_name = floor_ele->GetAttribute("name")->GetAsString();
       string model_name =
         floor_ele->GetAttribute("model_name")->GetAsString();
+
+      std::vector<string> models;
+      for (sdf::ElementPtr model_ele = floor_ele->GetFirstElement();
+        model_ele;
+        model_ele = model_ele->GetNextElement("model"))
+      {
+        if (model_ele->GetName() != string("model"))
```
Comment (Yadunund) [inline building_sim_plugins/building_gazebo_plugins/src/toggle_floors.cpp:54]: I think we can get away with checking for this if we change the structure of the loop.
```
auto model_ele = floor_ele->GetElement("model")
while (model_ele)
{
  if (model_ele.HasAttribute("name")
    models.push_back(model_ele->GetAttribute("name")->GetAsString());
  model_ele = model_ele->GetNextElement("model");
}
```

**Review by Yadunund — CHANGES_REQUESTED** [review]
Thanks for getting this out quickly! The approach is different from the one discussed in #219 but effective nonetheless. Lets reference this method and see what the best approach is. I've left a couple of minor comments for your consideration.

**Comment by Yadunund** [comment]
Just a thought, should we also toggle visibility of doors on the level?

**Comment by Kevinskwk** [comment]
I've implemented visibility toggling of doors (normal doors and lift shaft doors) on each level by appending the door model names in the same way. And I think it makes it look a lot better. While I don't think we should do the same to the non-static models. They are also hard to track since they might be moving between different levels.

**Comment by Kevinskwk** [comment]
I've implemented the toggling of the robot models. The plugin stores the current floor visibilities and robot visibilities and checks if a robot's visibility has changed by checking its current floor in the `RobotState` msg. It's similar to what's done in the `fleet_state_visualizer` and it's slightly laggy due to the `RobotState` msg publishing interval.

I haven't come up with a method to track the current floors of other non-static models (e.g. coke cans in delivery). Any idea on this topic is appreciated.

**Comment by tanyouliang95** [comment]
Tested this, and it works as expected! Thanks @Kevinskwk for the awesome work :tada:

**Review by Yadunund — APPROVED** [review]

**Pull request merged**